File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.cc
#include <stdio.h>

int main(void) {
    int n, m, i, j, k, l, temp = 0;
    scanf("%d %d", &n, &m);
    int arr[n];
    
    for(k = 1; k <= n; k++) {
        arr[k-1] = k;
    }

    for(k = 0; k < m; k++) {
        scanf("%d %d", &i, &j);
        for(l = 0; l < (j-i+1)/2; l++) {
            temp = arr[i+l-1];
            arr[i+l-1] = arr[j-l-1];
            arr[j-l-1] = temp;
        }
    }
    
    for(k = 0; k < n; k++) {
        printf("%d ", arr[k]);
    }
}

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        int m = scanner.nextInt();

        int[] arr = new int[n];
        int a,b,temp;

        for(a=1; a<=n; a++) {
            arr[a-1] = a;
        }

        for(a=0; a<m; a++) {
            int i = scanner.nextInt();
            int j = scanner.nextInt();
            for(b=0; b<(j-i+1)/2; b++) {
                temp = arr[i+b-1];
                arr[i+b-1] = arr[j-b-1];
                arr[j-b-1] = temp;
            }
        }

        for(a=0; a<n; a++) {
            System.out.printf(arr[a] + " ");
        }


    }
}